### PR TITLE
Remove biome users table

### PR DESCRIPTION
**Pull request by ltseeley**
Removes the Biome users table to match the most recent Biome design

### CODE REVIEW

File: libsplinter/src/biome/oauth/store/diesel/mod.rs [hunk 1]
```diff
@@ -293,8 +287,8 @@ pub mod tests {
     /// OAuthUser's `refresh_token`.
     ///
     /// 1. Create a connection pool for an in-memory SQLite database and run migrations.
-    /// 2. Create a `DieselUserStore` and a `DieselOAuthUserStore`.
-    /// 3. Add a User and an OAuthUser
+    /// 2. Createa `DieselOAuthUserStore`.
```
Comment (vaporos) [inline libsplinter/src/biome/oauth/store/diesel/mod.rs:290]: Typo - Createa

Comment (ltseeley) [reply libsplinter/src/biome/oauth/store/diesel/mod.rs]: Fixed

**Review by vaporos — APPROVED** [review]

File: splinterd/src/daemon.rs [hunk 2]
```diff
@@ -857,9 +856,10 @@ fn build_biome_routes(
     store_factory: &dyn splinter::store::StoreFactory,
 ) -> Result<BiomeRestResourceManager, StartError> {
     info!("Adding biome routes");
+    // Allowing unused_mut because `biome_rest_provider_builder` must be mutable if features
+    // biome-credentials or biome-key-management are enabled
+    #[allow(unused_mut)]
```
Comment (peterschwarz) [inline splinterd/src/daemon.rs:861]: Should we only create this if at least one of "biome-credentials" or "biome-key-management" is enabled?

Comment (ltseeley) [reply splinterd/src/daemon.rs]: Good idea; updated.

**Review by peterschwarz — DISMISSED** [review]

**Review by peterschwarz — DISMISSED** [review]

**Review by peterschwarz — APPROVED** [review]

**Review by vaporos — DISMISSED** [review]

**Review by peterschwarz — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**